packages/@webex/http-core/src/request/utils.ts

```diff
@@ -0,0 +1,63 @@
+/* eslint-disable import/prefer-default-export */
+import {EventEmitter} from 'events';
+import Interceptor from '../lib/interceptor';
+
+/**
+ * @param {Object} options
+ * @param {Array<Object>} interceptors
+ * @param {string} key
+ * @param {Object | undefined} res
+ * @private
+ * @returns {Promise}
+ */
+export function intercept(
+  options: object,
+  interceptors: Array<Interceptor>,
+  key: string,
+  res: object = undefined
+): Promise<any> {
+  const successKey = `on${key}`;
+  const errorKey = `on${key}Error`;
+
+  return interceptors.reduce(
+    (promise, interceptor) =>
+      promise.then(
+        (result) => {
+          interceptor.logOptions(options);
+          if (interceptor[successKey]) {
+            return interceptor[successKey](options, result);
+          }
+
+          return Promise.resolve(result);
+        },
+        (reason) => {
+          interceptor.logOptions(options);
+          if (interceptor[errorKey]) {
+            return interceptor[errorKey](options, reason);
+          }
+
+          return Promise.reject(reason);
+        }
+      ),
+    Promise.resolve(res)
+  );
+}
+
+/**
+ * Prepare options for a fetch.
+ * @param {object} options
+ * @returns {Promise}
+ */
+export async function prepareFetchOptions(options: any): Promise<any> {
```

Comment:
should this be named as prepateFetchPayload ?

Reply:
I don't think so, since this is more than the payload:
```
{
    "method": "POST",
    "headers": {
        "Content-Type": "application/json; charset=utf-8",
        "trackingid": "webex-web-client_4106f3bb-c618-4341-a358-02bc2fec1f8a_66",
        "spark-user-agent": "webex-js-sdk/3.0.0-beta.157 (web)",
        "authorization": "Bearer eyJhbGciOiJSUzI1NiJ9.eyJjbHVzdGVyIjoiUDBBMSIsInByaXZhdGUiOiJleUpqZEhraU9pSktWMVFpTENKbGJtTWlPaUpCTVRJNFEwSkRMVWhUTWpVMklpd2lZV3huSWpvaVpHbHlJbjAuLmMwcDE4WHpCbHBFbmxPakZ1dGZudWcuWlc3T3B6a29PbldsXzNMb2RqU0JuWlNnUlI0dHNFdDdDVTV6bC1XdV82UXl0eU5aMklTc2JJbEptSThQRzVGWDNHNXNqcllVRUFwdmE0NTgybjhPUmFNZGhLN204MFVkS0l1YV9MQjB3WmJYa1pFQXR4ZnVpUnFhQm5iS040RTNfblZKdGhQUEtJZ01jTWhRTWVBNWpES2hIM0g3Zk9VUFYwR04zdzdmQ0U0SmprT25GNmVNcmRSZGlIMVVraUdTdS1pMzd3WDI0ZExkbGJ3ZVlDR1hyMWdyZ1VmR1h0eTJrcG1Fb2haenVEazMyRDZiS3k5YkNldEpZXzR4ZFN3aVJOX1hJOWZpbWxTSEhCc2VtS0NHUFNKUzVCZ1ZIVTJZX0x1TF9vZHBXMVpOakxnMTRBQW44X0k4a0dhaUFUV053S1oxTXc4cEJQQWlOdDNmZ182em5fZnMzQ3JJUWVhNWtKdnpXb0hBWHRsajM3UmlmNkprWV8yS19GMmg2MDZJMUo4SGdfejZHczZBV3NjVVlIX29FMWM5TmxXMDBxWVlvOU9Hekx1VzUzQjdRanhTay1hQkZ0aDI3d2stUm1fWTVGXzdEWnkwV29oWnl0akFLVGFSYWRqbFYwQTA0dVhlU0JEb21ldEt4T1E3M2RIc2pFSGVJQ3lhZkpJV1RKaEoybmVLR1RaZ3dQN3AwSVN2bk9qcmtwcVdoQW1WZXp2X3lXX3Iwa2ZVbDFiNWw1NXotZ0lwM3NKaUU4VEFfVmtZWi00NWhER09jOFNzVno0UWZXa0VvZkMyOEZ2aFducmtGUk5XYm1IeHVHbWVnc3c2d1c5STBUNEtZNllZa1J1YlR3R3FtVkdadEV5ZW95OHpSRVBWc2FxcDBOT05EblJrTlNFZ3YxTXFDbE1Hclk1UGxHU21WM2JiNF9wc1M1cWpiZkQtUTl5aUtCZXI4WG91bENkc1RnQVZTSFB5dE14SHBTVFhuNjZFTFhpY2JNb3dsb2VqanctSzFpcFJ3ek9JNlhTYWxFTW9wV0VSZ0E5ZmZXa096Q0RxbllZaFZ6dGYwaFYtNzNQY2pLSS5MRFRBMHhvRzFxUWhXLXF1RXBwMU5nIiwidXNlcl90eXBlIjoidXNlciIsInRva2VuX2lkIjoiQWFaM3IwTmpWaE1ETXdaRFl0WldRNU9DMDBOREptTFdFMk5EVXRNVFJsT1dFeU9HWXlOR0prTVRRME9EUTBNR0V0T1dZMSIsInJlZmVyZW5jZV9pZCI6IjhiMzc4NGYzLWIxNGYtNDZiMC04ZjExLTBiMjhjZDcxNGU3MiIsImlzcyI6Imh0dHBzOlwvXC9pZGJyb2tlci1iLXVzLndlYmV4LmNvbVwvaWRiIiwidXNlcl9tb2RpZnlfdGltZXN0YW1wIjoiMjAyMjA3MjcxMjA5MzEuNDM1WiIsInJlYWxtIjoiMTkwNjczMzMtNGY3Yy00MWRkLWIyNTktODA2YTNlOTRmMmFjIiwiY2lzX3V1aWQiOiJlY2IyNzljNS0yNDYyLTQ5YzEtYjVjNC02OWY3NzY4MzY3ZmMiLCJ0b2tlbl90eXBlIjoiQmVhcmVyIiwiZXhwaXJ5X3RpbWUiOjE2ODg2Mjk0NDc1NDEsImNsaWVudF9pZCI6IkM2NGFiMDQ2MzllZWZlZTQ3OThmNThlN2JjM2ZlMDFkNDcxNjFiZTBkOTdmZjBkMzFlMDQwYTZmZmU2NmQ3ZjBhIn0.kn3TMkZNsmlxPcf99TO1C9ArPLqu_6vJk06ApvpqUNzpcmkcrhNKb_1VgtT7DI6sbx97g8sck8K8XmCZYNJtKpX0u9L8wCGLBUL9ecj7Xx2wzcc4fNGukwOMdr0nEzPiIapjVmlKo34FXbiIRvlTmcF9n8IGQjsomWSSza0VFjEjZjJJnsq1R3qCQFhFf9miwacmIazoMUSUcZTqbMYIbnq-MwQNTkODWFeOoIatde8lFZFo9zVAnwtFNX4_WFgHPLhUaq2WZbZUBtaChXZMUh8k2_hJS0cfRvvfHu1FUlItYxGhP8uVlfN1yrr042js1DivcrF1CRuHOePhY12wTA",
        "cisco-no-http-redirect": true,
        "cisco-device-url": "https://wdm-r.wbx2.com/wdm/api/v1/devices/69d8a2b8-36b1-477b-91f1-7f50ce7cc775"
    },
    "name": "app_closed_with_active_survey",
    "payload": {
        "type": "operational",
        "tags": {
            "appversion": "v-1",
            "surveySessionId": "2785a5ae-42fa-40ff-8a95-c7653b03cb22"
        },
        "fields": {}
    },
    "type": "operational",
    "body": "{\"metrics\":[{\"metricName\":\"app_closed_with_active_survey\",\"tags\":{\"appversion\":\"v-1\",\"surveySessionId\":\"2785a5ae-42fa-40ff-8a95-c7653b03cb22\",\"browser\":\"Chrome\",\"os\":\"windows\",\"domain\":\"localhost\"},\"fields\":{\"browser_version\":\"114.0.0.0\",\"os_version\":\"NT 10.0\",\"sdk_version\":\"3.0.0-beta.157\",\"platform\":\"Web\",\"spark_user_agent\":\"webex-js-sdk\",\"client_id\":\"C64ab04639eefee4798f58e7bc3fe01d47161be0d97ff0d31e040a6ffe66d7f0a\"},\"type\":\"operational\",\"context\":{\"app\":{\"version\":\"3.0.0-beta.157\"},\"locale\":\"en-US\",\"os\":{\"name\":\"windows\",\"version\":\"NT 10.0\"}},\"timestamp\":1688565303196}]}",
    "api": "metrics",
    "resource": "clientmetrics",
    "keepalive": true,
    "url": null,
    "json": true,
    "uri": "https://metrics-r.wbx2.com/metrics/api/v1/clientmetrics",
    "$timings": {
        "requestStart": 1688565303215,
        "networkStart": 1688565303218
    },
    "$redirectCount": 0
}